src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,505 @@
+                individualOverlapInfo.numOverlap, individualOverlapInfo.overlapMappingQuality,
+                individualOverlapInfo.meanOverlapMappingQuality, individualOverlapInfo.numCoherent,
+                individualOverlapInfo.coherentMappingQuality,
+                clusterOverlapInfo.numOverlap, clusterOverlapInfo.overlapMappingQuality,
+                clusterOverlapInfo.meanOverlapMappingQuality, clusterOverlapInfo.numCoherent,
+                clusterOverlapInfo.coherentMappingQuality,
+                referenceGapOverlap, umapS100
+            }
+        );
+    }
+
+    private int getMappingQuality(final BreakpointEvidence evidence) {
+        // Note: return "max" mapping quality for non-ReadEvidence. Reasoning: some features depend on sum or average of
+        // read qualities. Non-ReadEvidence isn't *bad* per se, so give it a good score.
+        return evidence.getMappingQuality() == null ? 60: evidence.getMappingQuality();
+    }
+
+    private double getTemplateSizeOrReadCount(final BreakpointEvidence evidence) {
+        final Integer templateSize = evidence.getTemplateSize();
+        if(templateSize == null) {
+            // For TemplateSizeAnomaly, return readCount scaled by meanGenomeCoverage
+            final Integer readCounts = evidence.getReadCount();
+            if(readCounts == null) {
+                throw new IllegalStateException("templateSizeOrReadCount feature is only defined for ReadEvidence and TemplateSizeAnomaly, not "
+                        + evidence.getClass().getName());
+            }
+            return (double)(readCounts) / readMetadata.getCoverage();
+
+        } else {
+            // For ReadEvidence, return templateSize as percentile of library's cumulative density function:
+            final String readGroup = ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup();
+            final String library = readMetadata.getReadGroupToLibraryMap().get(readGroup);
+            final LibraryStatistics libraryStatistics = readMetadata.getLibraryStatistics(library);
+            final IntHistogram.CDF templateSizeCDF = libraryStatistics.getCDF();
+            final int cdfBin = Integer.min(Math.abs(templateSize), templateSizeCDF.size() - 1);
+            return templateSizeCDF.getFraction(cdfBin);
+        }
+    }
+
+    private CoverageScaledOverlapInfo getIndividualOverlapInfo(final BreakpointEvidence evidence) {
+        // Since overlap info will be needed for the same evidence in different contexts, it's fastest to calculate it
+        // once, cache it, and then just retrieve the info each time it's needed.
+        if(!rawFeatureCache.containsKey(evidence)) {
+            cacheOverlapInfo(evidence);
+        }
+        final UnscaledOverlapInfo evidenceFeatureCache = rawFeatureCache.get(evidence);
+        // Calculate the coverage scaled overlap info
+        final double meanOverlapMappingQuality = ((double)evidenceFeatureCache.overlapMappingQuality)
```

Comment:
See my comment below about maybe computing the mean in the overlap object instead of here.